### PR TITLE
Bump: AGP 7.1 beta and 7.2 alpha

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
_#skip-changelog_

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

**Review by romtsn — APPROVED** [review]

**Review by cortinico — APPROVED** [review]

File: .github/workflows/test-publish-dry-run.yaml
Comment (marandaneto) [inline .github/workflows/test-publish-dry-run.yaml:59]: for some reason getting:

> A problem occurred evaluating project ':examples:android-gradle'.
> > Failed to apply plugin 'com.android.internal.version-check'.
>    > Minimum supported Gradle version is 7.3-rc-1. Current version is 7.2. If using the gradle wrapper, try editing the distributionUrl in /home/runner/work/sentry-android-gradle-plugin/sentry-android-gradle-plugin/gradle/wrapper/gradle-wrapper.properties to gradle-7.3-rc-1-all.zip

**Comment by marandaneto** [comment]
I'd not merge this now because it requires Gradle 7.3 RC otherwise the `gradle wrapper` cmd fails due to https://github.com/getsentry/sentry-android-gradle-plugin/pull/230#discussion_r745404682
once 7.3 is GA, we can update this PR and merge it, but its good to know that CI is green

File: .github/workflows/test-publish-dry-run.yaml
Comment (cortinico) [reply .github/workflows/test-publish-dry-run.yaml:59]: Because AGP 7.2 requires Gradle 7.3 (to run any command, even `gradlew wrapper`). That's a hard requirement

**Comment by marandaneto** [comment]
> I'd not merge this now because it requires Gradle 7.3 RC otherwise the `gradle wrapper` cmd fails due to [#230 (comment)](https://github.com/getsentry/sentry-android-gradle-plugin/pull/230#discussion_r745404682) once 7.3 is GA, we can update this PR and merge it, but its good to know that CI is green

7.3.1 is GA, we can unblock this PR

**Comment by cortinico** [comment]
> 7.3.1 is GA, we can unblock this PR.

Yup this should be mergeable now

File: plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryPluginVariantTest.kt
Comment (marandaneto) [inline plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryPluginVariantTest.kt:135]: @romtsn Why did we remove that? Now we've lost the ability to run across the matrix for this test.

Comment (romtsn) [reply plugin-build/src/test/kotlin/io/sentry/android/gradle/SentryPluginVariantTest.kt:135]: it's a leftover from the past, this lives now in [here](https://github.com/getsentry/sentry-android-gradle-plugin/blob/d14f2b6f439c96bf227e73096a75b057fb365df2/plugin-build/src/test/kotlin/io/sentry/android/gradle/BaseSentryPluginTest.kt#L86-L110)

That's is still run parameterized, you can check the latest [CI run ](https://github.com/getsentry/sentry-android-gradle-plugin/actions/runs/1696777183)

**Pull request merged**
